src/frontend/src/optimizer/plan_node/logical_filter.rs

```diff
@@ -236,7 +236,11 @@ impl ToStream for LogicalFilter {
                             if try_match_expand!(&now_expr.inputs()[0], ExprImpl::FunctionCall)?
                                 .get_expr_type()
                                 != Type::Now
-                                || !matches!(&now_expr.inputs()[1], ExprImpl::Literal(_))
+                                || !matches!(
+                                    &now_expr.inputs()[1],
+                                    ExprImpl::Literal(_) | ExprImpl::FunctionCall(_)
+                                )
+                                || now_expr.inputs()[1].has_input_ref()
                             {
                                 return Err(ExprError::InvalidParam {
                                     name: "now",
```

Comment:
error msg here should be adjusted accordingly

Reply:
Hmm. Can we say "delta expression cannot reference any columns"?

Reply:
Actually, I don't want to confuse the user. I think the general use-case is to use a literal like '1 hr'. So I suggest we stick with this msg.